server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -0,0 +1,349 @@
+        return false;
+    }
+
+    @Override
+    public void syncTranslog() {
+    }
+
+    @Override
+    public Closeable acquireRetentionLockForPeerRecovery() {
+        return () -> {};
+    }
+
+    @Override
+    public Translog.Snapshot newChangesSnapshot(String source, MapperService mapperService, long fromSeqNo, long toSeqNo,
+                                                boolean requiredFullRange) throws IOException {
+        return readHistoryOperations(source, mapperService, fromSeqNo);
+    }
+
+    @Override
+    public Translog.Snapshot readHistoryOperations(String source, MapperService mapperService, long startingSeqNo) throws IOException {
+        return new Translog.Snapshot() {
+
+            @Override
+            public void close() { }
+
+            @Override
+            public int totalOperations() {
+                return 0;
+            }
+
+            @Override
+            public Translog.Operation next() {
+                return null;
+            }
+        };
+    }
+
+    @Override
+    public int estimateNumberOfHistoryOperations(String source, MapperService mapperService, long startingSeqNo) throws IOException {
+        return 0;
+    }
+
+    @Override
+    public boolean hasCompleteOperationHistory(String source, MapperService mapperService, long startingSeqNo) throws IOException {
+        return false;
+    }
+
+    @Override
+    public TranslogStats getTranslogStats() {
+        return translogStats;
+    }
+
+    @Override
+    public Translog.Location getTranslogLastWriteLocation() {
+        return new Translog.Location(0,0,0);
+    }
+
+    @Override
+    public long getLocalCheckpoint() {
+        return seqNoStats.getLocalCheckpoint();
+    }
+
+    @Override
+    public void waitForOpsToComplete(long seqNo) { }
+
+    @Override
+    public void resetLocalCheckpoint(long newCheckpoint) {
+    }
+
+    @Override
+    public SeqNoStats getSeqNoStats(long globalCheckpoint) {
+        return new SeqNoStats(seqNoStats.getMaxSeqNo(), seqNoStats.getLocalCheckpoint(), globalCheckpoint);
+    }
+
+    @Override
+    public long getLastSyncedGlobalCheckpoint() {
+        return seqNoStats.getGlobalCheckpoint();
+    }
+
+    @Override
+    public long getIndexBufferRAMBytesUsed() {
+        return 0;
+    }
+
+    @Override
+    public List<Segment> segments(boolean verbose) {
+        return Arrays.asList(getSegmentInfo(lastCommittedSegmentInfos, verbose));
+    }
+
+    @Override
+    public void refresh(String source) throws EngineException {}
+
+    @Override
+    public void writeIndexingBuffer() throws EngineException {}
+
+    @Override
+    public boolean shouldPeriodicallyFlush() {
+        return false;
+    }
+
+    @Override
+    public SyncedFlushResult syncFlush(String syncId, CommitId expectedCommitId) throws EngineException {
+        // we can't do synced flushes this would require an indexWriter which we don't have
+        return SyncedFlushResult.COMMIT_MISMATCH;
```

Comment:
hmm... I really wonder if we are better off throwing in an exception here...

Reply:
yeah lets do that. I think we can cope with it.